gpflow/models/gplvm.py

```diff
@@ -145,7 +149,8 @@ def __init__(
         assert self.X_prior_var.shape[0] == self.num_data
         assert self.X_prior_var.shape[1] == self.num_latent_gps
 
-    def maximum_log_likelihood_objective(self) -> tf.Tensor:
+    # type-ignore is because of changed method signature:
+    def maximum_log_likelihood_objective(self) -> tf.Tensor:  # type: ignore
```

Comment:
Why does this have an incompatible method? Doesn't that cause issues? Is it basically the case that you can only call `maximum_log_likelihood_objective` if you know the precise type of GPModel?

Presumably the abstract `maximum_log_likelihood_objective` and `_training_loss` methods include args and kwargs parameters to suggest that submethods might expect additional arguments?

Reply:
> Why does this have an incompatible method? Doesn't that cause issues? Is it basically the case that you can only call `maximum_log_likelihood_objective` if you know the precise type of GPModel?
> 
> Presumably the abstract `maximum_log_likelihood_objective` and `_training_loss` methods include args and kwargs parameters to suggest that submethods might expect additional arguments?

Yes, it's because you can pass arbitrary `*args` and `**kwargs` in the super class. Yes, that violates the "Liskov Substitution Principle", which isn't great. But this isn't the time and place to work on that.